### PR TITLE
[14.0][ADD] Addon: sale_order_sign_oca

### PR DESCRIPTION
**Pull request by Tiago370**
Sign module for the sale order.

### CODE REVIEW

**Comment by mileo** [comment]
@Tiago370 could fix the readme files?

**Review by mileo — APPROVED** [review]

File: sale_order_sign_oca/views/sign_oca_request.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 KMEE
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record model="ir.ui.view" id="sign_oca_request_search_view">
+        <field name="name">sign.oca.request.search (in sale_order_sign_oca)</field>
+        <field name="model">sign.oca.request</field>
+        <field name="inherit_id" ref="sign_oca.sign_oca_request_search_view" />
```
Comment (victoralmau) [inline sale_order_sign_oca/views/sign_oca_request.xml:9]:
```suggestion
        <field name="inherit_id" ref="sign_oca.sign_oca_request_search_view" />
        <field
            name="groups_id"
            eval="[(6, 0, [ref('sales_team.group_sale_salesman')])]"
        />

```

File: sale_order_sign_oca/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                sale_order._generate_sign_oca_request()
+        return res
+
+    def write(self, vals):
+        old_partner_id = self.partner_id
+        new_partner_id = vals.get("partner_id")
+        res = super().write(vals)
+        if new_partner_id and new_partner_id != old_partner_id:
+            self._generate_sign_oca_request()
+        return res
```
Comment (victoralmau) [inline sale_order_sign_oca/models/sale_order.py]: Although the flow is correct, I am not sure that this is “_necessary_” or that you always want to do this (create and send a signature request when creating a quote, perhaps when sending it by email). 

One possibility would be to add to res.config.settings the ability to select the order states in which you want to auto-create.

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code and functional review OK

**Comment by Tiago370** [comment]
@victoralmau It is now possible to select in which state of the sales order the subscription request will be automatically generated, as you suggested.

@mileo

File: sale_order_sign_oca/views/res_config_settings.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+<!-- <div class="o_setting_right_pane">
+    <span class="o_form_label">Confirmation Email</span>
+    <div class="text-muted">
+                                    Automatic email sent after the customer has signed or paid online
+    </div>
+    <div class="row mt16">
+        <label for="confirmation_template_id" string="Email Template" class="col-lg-4 o_light_label"/>
+        <field name="confirmation_template_id" class="oe_inline"/>
+    </div>
+</div> -->
```
Comment (victoralmau) [inline sale_order_sign_oca/views/res_config_settings.xml]:
```suggestion
```

File: sale_order_sign_oca/views/res_config_settings.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                                        class="btn-link"
+                                    />
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                    <div
+                        class="col-12 col-lg-6 o_setting_box"
+                        id="invoicing_policy_setting"
+                        title="The mode selected here applies as invoicing policy of any new product created but not of products already existing."
```
Comment (victoralmau) [inline sale_order_sign_oca/views/res_config_settings.xml]:
```suggestion
                        id="div_sale_order_sign_oca_state"
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code review OK

**Comment by Tiago370** [comment]
@victoralmau everything fixed.

**Review by victoralmau — APPROVED** [review]
Code review OK

**Review by rvalyi — APPROVED** [review]

**Pull request merged**